### PR TITLE
All day events

### PR DESCRIPTION
**Pull request by b-forster**
Filter out all-day events when the current time is past 6 pm; refactor #time_until method

### CODE REVIEW

**Comment by b-forster** [comment]
This passes all rspec tests locally, but Travis is failing due to a bug:

**Comment by b-forster** [comment]
https://github.com/travis-ci/travis-ci/issues/7685

File: app/helpers/alexa_interface_helper.rb [hunk 1]
```diff
@@ -31,16 +31,16 @@ def call(call_parameters={})
     response["events"]["event"]
   end
 
-  #limit the selection to events that have not yet started or are all-day events
+  #limit the selection to events that have not yet started (OR all-day events if the current time is late in the day)
```
Comment (namuunaa) [inline app/helpers/alexa_interface_helper.rb]: in this comment, the all day events should be early in the day, right?

File: app/helpers/alexa_interface_helper.rb [hunk 2]
```diff
@@ -31,16 +31,16 @@ def call(call_parameters={})
...
-    call_list = call_list.select do |event|
-      event["all_day"] && event["start_time"]
-    end
+    # call_list = call_list.select do |event|
+    #   event["all_day"] && event["start_time"]
+    # end
     call_list = call_list.select do |event|
       Time.zone = event['olson_path']
-      event["all_day"] != "0" || Time.zone.parse(event["start_time"]).future?
+      event["all_day"] != "0" && Time.now.strftime('%R') >= "18:00" || Time.zone.parse(event["start_time"]).future?
```
Comment (namuunbayaraa) [inline app/helpers/alexa_interface_helper.rb]: switch to <= 18:00

File: spec/helpers/alexa_interface_helper_spec.rb [hunk 3]
```diff
@@ -21,19 +21,49 @@
...
-      response.each do |event|
+      api_response.each do |event|
         event_start = Time.parse(event["start_time"])
-        event_not_started = (event_start > Time.now || event['all_day'] != 0 )
+        event_not_started = (event_start > Time.now || event['all_day'] != "0" )
         expect(event_not_started).to be true
       end
     end
 
+    it 'selects all-day events if the current time is after 6 pm' do
```
Comment (namuunbayaraa) [inline spec/helpers/alexa_interface_helper_spec.rb]: switch to before 6pm

File: spec/helpers/alexa_interface_helper_spec.rb [hunk 4]
```diff
@@ -21,19 +21,49 @@
...
+
+      allow(Time).to receive(:now).and_return(current_time)
+
+      all_day_test_response.each do |event|
+
+        expect(all_day_test_response.length).to be 3
+      end
+    end
 
+    it 'does not select all-day events if the current time is 6pm or later' do
```
Comment (namuunbayaraa) [inline spec/helpers/alexa_interface_helper_spec.rb]: switch to before 6pm

File: app/helpers/alexa_interface_helper.rb (hunk 1 above)
Comment (namuunbayaraa) [inline app/helpers/alexa_interface_helper.rb:34]: in this comment, the all day events should be early in the day, right?

**Review by namuunbayaraa — CHANGES_REQUESTED** [review]

**Review by namuunbayaraa — APPROVED** [review]

**Pull request merged**